litmus-portal/graphql-server/pkg/analytics/ops/operations.go

```diff
@@ -216,3 +218,26 @@ func MapMetricsToDashboard(dashboardQueryMap []*model.QueryMapForPanelGroup, new
 
 	return dashboardResponse
 }
+
+// UpdateViewedAt updates the viewed_at field of a dashboard based on dashboard id and it's view id
+func UpdateViewedAt(dashboardID *string, viewID string) error {
+	if dashboardID != nil && *dashboardID != "" {
+		timestamp := strconv.FormatInt(time.Now().Unix(), 10)
+		query := bson.D{
+			{"db_id", dashboardID},
+			{"is_removed", false},
+		}
+		update := bson.D{{"$set", bson.D{{"viewed_at", timestamp}}}}
+		err := dbOperationsAnalytics.UpdateDashboard(query, update)
+		if err != nil {
+			log.Printf("error updating viewed_at field of the dashboard: %v\n", *dashboardID)
```

Comment:
Instead of printing the err. do return errors.New(). Same with other places

Reply:
updating this to not return any error as we are not handling this error in frontend and its not used because this happens at the closing of subscription when user leaves the dashboard page so just log was needed here, its a non critical error. @rajdas98